src/lib/ledger_builder_controller/transition_logic.ml

```diff
@@ -230,66 +230,43 @@ module Make (Inputs : Inputs_intf) :
 
     let run old_state ~on_success new_tree old_tree new_best_path logger
         _transition =
-      let locked_tip = Transition_logic_state.locked_tip old_state
-      and longest_branch_tip =
-        Transition_logic_state.longest_branch_tip old_state
-      in
-      let new_head, _new_tip = locked_and_best new_tree in
-      let old_head, _old_tip = locked_and_best old_tree in
-      let open Interruptible.Let_syntax in
-      let ivar : (External_transition.t, State_hash.t) With_hash.t Ivar.t =
-        Ivar.create ()
-      in
-      let step tip_with_hash transition_with_hash =
-        Interruptible.lift
-          (Step.step logger tip_with_hash transition_with_hash)
-          (Deferred.map (Ivar.read ivar) ~f:ignore)
-      in
-      let work =
-        let locked_tip = With_hash.map locked_tip ~f:Tip.copy in
-        let longest_branch_tip = With_hash.map longest_branch_tip ~f:Tip.copy in
-
-        (* Adjust the locked_ledger if necessary *)
-        let%bind locked_tip =
-          if transition_is_parent_of ~child:new_head ~parent:old_head then
-            transition_unchecked locked_tip.data new_head logger
-          else return locked_tip
-        in
-        (* Now adjust the longest_branch_tip *)
-        let tip, path =
-          match
-            Path.findi new_best_path ~f:(fun _ x ->
-                is_materialization_of longest_branch_tip x )
-          with
-          | None -> (locked_tip, new_best_path)
-          | Some (i, _) ->
-              ( longest_branch_tip
-              , Path.drop new_best_path (i + 1) )
-        in
-        let last_transition = List.last_exn path.Path.path in
-        (* Now step over the path *)
-        assert (is_materialization_of tip path.Path.source) ;
```

Comment:
what happened to this code?

Reply:
Looks like I messed up the merge with Corey's tracking branch. Will fix this now.

Reply:
Actually, it seems to be on par for master.
https://github.com/CodaProtocol/coda/blob/master/src/lib/ledger_builder_controller/transition_logic.ml#L234